core/trino-main/src/main/java/io/trino/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -446,26 +473,32 @@ public void addInput(GroupByIdBlock groupIdsBlock, Page page)
         }
 
         @Override
-        public void addIntermediate(GroupByIdBlock groupIdsBlock, Block block)
+        public void evaluateFinal(int groupId, BlockBuilder output)
         {
-            throw new UnsupportedOperationException();
+            accumulator.evaluateFinal(groupId, output);
         }
 
         @Override
-        public void evaluateIntermediate(int groupId, BlockBuilder output)
+        public void prepareFinal()
+        {
+        }
+
+        @Override
+        public final Type getIntermediateType()
         {
             throw new UnsupportedOperationException();
         }
 
         @Override
-        public void evaluateFinal(int groupId, BlockBuilder output)
+        public final void addIntermediate(GroupByIdBlock groupIdsBlock, Block block)
```

Comment:
`final` is unnecessary here. The class is and internal implementation detail of GenericAccumulatorFactory and not meant to be extended. If anything, the whole class should be marked as final.